src/consensus/malachite/read_host.rs

```diff
@@ -70,7 +72,11 @@ impl ReadHost {
             }
 
             ReadHostMsg::ProcessDecidedValue { value, sync } => {
-                state.validator.process_decided_value(value, &sync);
+                let num_values_processed = state.validator.process_decided_value(value);
+                if num_values_processed > 0 {
+                    sync.cast(read_sync::Msg::Decided(state.validator.last_height))
+                        .unwrap();
```

Comment:
These unwraps are fine since it's just a read node, but we have to be careful about panics. It's better to use `?` and bubble up the error

Reply:
Fixed